### PR TITLE
Backend validation-PAY-5314

### PR DESCRIPTION
**Pull request by ranjeet-kumarhmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PAY-5314

### Change description ###

Backend : Business rule validation for refund

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: charts/payment-api/values.yaml [hunk 1]
```diff
@@ -81,6 +81,12 @@ java:
     REFUND_API_URL: http://ccpay-refunds-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
     SERVER_SHUTDOWN_STRATEGY: graceful
     TIMEOUT_SERVER_SHUTDOWN_PHASE: 30s
+    IDAM_API_URL: "https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net"
```
Comment (ranjeet-kumarhmcts) [inline charts/payment-api/values.yaml:84]: remove above line

File: model/src/test/resources/application-test.properties [hunk 2]
```diff
@@ -3,3 +3,11 @@ launchDarkly.connectionTimeout=3
...
 launchDarkly.flushInterval=10
 launch.darkly.user.name="testUser"
+
+card.lag.time=120
+cash.lag.time=120
+postalorders.lag.time=480
+cheques.lag.time=480
+pba.lag.time=96
+
+idam.api.url=${IDAM_API_URL:http://localhost:5000}
```
Comment (ranjeet-kumarhmcts) [inline model/src/test/resources/application-test.properties:13]: Rename idam.api.url as per values.yaml

File: api/src/main/java/uk/gov/hmcts/payment/api/service/RefundRemissionEnableServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        if(!roles.getRoles().isEmpty()){
+            isRoles = checkRoles(roles.getRoles());
+        }
+
+        Optional<Remission> remission = remissionRepository.findByFeeId(fee.getId());
+        if(remission.isPresent()) {
+            isRemission=true;
+        }
+
+        boolean refundEnableFeature = featureToggler.getBooleanValue("refund-remission-feature",false);
```
[the remark this answers is not in the document]
Comment (ranjeet-kumarhmcts) [reply api/src/main/java/uk/gov/hmcts/payment/api/service/RefundRemissionEnableServiceImpl.java]: updated flag name and boolean

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/RefundsRequestorJourneyFunctionalTest.java [hunk 4]
```diff
@@ -76,13 +81,19 @@
     @Autowired
     private PaymentsTestDsl dsl;
 
+    /*@Autowired
+    @Qualifier("paymentServiceImpl")
```
Comment (YasharRahvar) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/RefundsRequestorJourneyFunctionalTest.java:85]: remove comments if not longer needed

**Pull request merged**